### ISSUE TITLE
functions in extend is not extended in types

### ISSUE THREAD

**Issue by AdrienLemaire** [muhammadsammy/tailwindcss-classnames#49]
Following the [extending the default theme](https://tailwindcss.com/docs/theme#extending-the-default-theme) doc, I changed my settings as follow:

```diff
   theme: {
-    boxShadow: (theme) => ({
-      default:
-        "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)",
-      md:
-        "0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)",
-      xl:
-        "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
-      outline: `0 0 0 3px ${theme("colors.black.alpha")}`,
-      goblue: `0 0 0 3px ${theme("colors.goblue.alpha")}`,
-      gopink: `0 0 0 3px ${theme("colors.gopink.alpha")}`,
-    }),
+    extend: {
+      boxShadow: (theme) => ({
+        outline: `0 0 0 3px ${theme("colors.black.alpha")}`,
+        goblue: `0 0 0 3px ${theme("colors.goblue.alpha")}`,
+        gopink: `0 0 0 3px ${theme("colors.gopink.alpha")}`,
+      }),
+    },
}
```

I'm expecting the types to be:
```ts
export type TBoxShadow = 
  | 'shadow'
  | 'shadow-md'
  | 'shadow-xl'
  | 'shadow-outline'
  | 'shadow-goblue'
  | 'shadow-gopink';
```

But I'm getting
```ts
export type TBoxShadow = 
  | 'shadow-outline'
  | 'shadow-goblue'
  | 'shadow-gopink';
```

**Comment by muhammadsammy** [comment]
I tried this with `boxShadow` as an object and it works fine, for some reason when using a property as a function in `extend` it does not extend the property but overrides it

**Comment by muhammadsammy** [comment]
fixed by 3be8b6f
